### DISCUSSION TITLE
How about a Datenstrom Yellow package for YunoHost?

### DISCUSSION THREAD

**Discussion by nibreh** [datenstrom/community#88] [Ask a question] [answered]
Hi

I would like to use Yellow with [YunoHost](https://yunohost.org/#/) (https://en.wikipedia.org/wiki/YunoHost). 

Yunohost come with nginx. Every app on Yunohost have is own nginx configuration. I use a special "yunohost app" to install any custom app or html pages in a folder (it's like an empty folder):

I can't use the configuration for nginx like in the wiki https://github.com/datenstrom/yellow/wiki/Web-server-configuration

This is the default nginx conf for the _virtual host_ where I want to install Yellow

```
location /home {
   alias /var/www/webapp_name/home/;
   if ($scheme = http) {
          rewrite ^ https://$server_name$request_uri? permanent;
   }
   index index.php index.html index.htm;
   default_type text/html;
   location ~ [^/]\.php(/|$) {
       fastcgi_split_path_info ^(.+?\.php)(/.*)$;
       fastcgi_pass unix:/var/run/php5-fpm.sock;
       fastcgi_index index.php;
       include fastcgi_params;
       fastcgi_param  REMOTE_USER $remote_user;
       fastcgi_param  PATH_INFO $fastcgi_path_info;
    fastcgi_param SCRIPT_FILENAME $request_filename;
   }
  # Include SSOWAT user panel.
  include conf.d/yunohost_panel.conf.inc;
}
```

Maybe someone can help me and told me how to include yellow nginx config in it ? 

I think about @dcschmid (https://github.com/datenstrom/yellow/issues/18)

**Comment by nibreh** [comment]
How to install Yellow in YunoHost:

1-Install [Custom Webapp without FTP access](https://github.com/maniackcrudelis/my_webapp_ynh) with yunohost (for example, in **domain.com/home**)
2-Upload Yellow in `/var/www/webapp_name/home` (via SSH) (_name_ is your username, _home_ is the folder)
3-Edit **home.conf** (or _something.conf_) in folder `/etc/nginx/conf.d/domain.com.d` like this:  

```
error_page 404 /home/error.html;

location /home/cache {
    rewrite ^/home/(.*)$ /home/error break;
}
location /home/content {
    rewrite ^/home/(.*)$ /home/error break;
}
location /home/system {
    rewrite ^/home/(.*)$ /home/error break;
}

location /home {
     alias /var/www/webapp_name/home/;
     if ($scheme = http) {
          rewrite ^ https://$server_name$request_uri? permanent;
    }
     index yellow.php index.php index.html index.htm;
     default_type text/html;
     location ~ [^/]\.php(/|$) {
       fastcgi_split_path_info ^(.+?\.php)(/.*)$;
       fastcgi_pass unix:/var/run/php5-fpm.sock;
       fastcgi_index yellow.php;
       include fastcgi_params;
       fastcgi_param  REMOTE_USER $remote_user;
       fastcgi_param  PATH_INFO $fastcgi_path_info;
       fastcgi_param SCRIPT_FILENAME $request_filename;
    }

    if (!-e $request_filename) {
    rewrite ^/home/(.*)$ /home/yellow.php last;
    break;
    }

# Include SSOWAT user panel.
include conf.d/yunohost_panel.conf.inc;
}
```

4-Edit **config.ini** in `/var/www/webapp_name/home/system/config` and add this:

`webinterfaceServerScheme = https` 

Working good for now.

_This is only work for [my_webapp without FTP access](https://github.com/maniackcrudelis/my_webapp_ynh), I have not tested yet with official my_webapp (https://github.com/Kloadut/my_webapp_ynh)_

**Comment by nibreh** [comment]
Now that we know how to configure nginx, maybe someone will packaged Yellow for Yunohost or help me to do it ?

English Tutorial:
https://yunohost.org/#/packaging_apps_en

French Tutorial:
https://yunohost.org/#/packaging_apps_fr

**Comment by markseuffert** [comment]
There hasn't been any comment for a while, do you want to keep this issue open?

**Comment by nibreh** [comment, marked as answer]
In fact, a yellow package for yunohost is not urgent. A package is just a more easier installation for webapps in YunoHost. But **if you follow the tip above, Yellow it's working fine with Yunohost 2.2**

In my opinion, **if there is a strong demand from YunoHost's users**, we can make a package for Yellow.

Furthermore, [Yunohost 2.4 is coming soon](https://dev.yunohost.org/projects/yunohost/roadmap). 

I suggest to close this issue and maybe wait for the new release.

_More tips for YunoHost: https://yunohost.org_

**Reply by markseuffert** [reply]
A Yellow package has been added, see https://github.com/YunoHost-Apps/yellow_ynh.
